server.go

```diff
@@ -179,6 +173,7 @@ type serverOptions struct {
 }
 
 var defaultServerOptions = serverOptions{
+	maxConcurrentStreams:  math.MaxUint32,
```

Comment:
Is our interpretation of the spec correct in setting the default to `math.MaxUint32`?

The line I'm talking about specifically is this: https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2
`Initially, there is no limit to this value.`